src/buddy.php

```diff
@@ -1,83 +1,74 @@
 <?php declare(strict_types=1);
 
-include 'utils.php';
+namespace Manticoresearch\Buddy;
 
+use Manticoresearch\Buddy\Exception\BuddyRequestError;
+use Manticoresearch\Buddy\Exception\ParserLocationError;
+use Manticoresearch\Buddy\Exception\SocketError;
+use Manticoresearch\Buddy\Lib\Buddy;
+use Manticoresearch\Buddy\Lib\QueryParserLocator;
+use Manticoresearch\Buddy\Lib\SocketHandler;
+use Manticoresearch\Buddy\Lib\StatementBuilder;
+use RuntimeException;
+use Throwable;
 
 $longopts  = [
 	'pid:',
 	'pid_path:',
+	'listen:',
+	'sockAddr::',
+	'sockPort::',
 ];
-/** @var string $pid */
-/** @var string $pidPath */
-// @phpstan-ignore-next-line
-['pid' => $pid, 'pid_path' => $pidPath] = getopt('', $longopts);
-
-const ACTION_MAP = [
-	'NO_INDEX' => [
-		'INSERT_QUERY' => 'CREATE_INDEX',
-	],
-	'UNKNOWN_COMMAND' => [
-		'SHOW_QUERIES_QUERY' => 'SELECT_SYSTEM_SESSIONS',
-	],
-];
-
-const ENDPOINT_MAP = [
-	'/sql?mode=raw' => ['CREATE_INDEX'],
-	'/sql' => ['SELECT_SYSTEM_SESSIONS'],
+$defaultOpts = [
+	'sockAddr' => '127.0.0.1',
+	'sockPort' => 0,
 ];
+/** @var string $clPid */
+/** @var string $clPidPath */
+/** @var string $clAddrPort */
+/** @var string $sockAddr */
+/** @var string $sockPort */
+[
+	'pid' => $clPid,
+	'pid_path' => $clPidPath,
+	'listen' => $clAddrPort,
+	'sockAddr' => $sockAddr,
+	'sockPort' => $sockPort,
+] = getopt('', $longopts) + $defaultOpts;
 
-// bundles are sorted by type priority from the highest to the lowest
-const TYPE_BUNDLES = [
-	['json'],
-	['multi64', 'multi'],
-	['float', 'bigint', 'int'],
-	['text', 'string'],
-];
+$sockPort = (int)$sockPort;
+$socketHandler = new SocketHandler($sockAddr, $sockPort, new SocketError());
+$parserLocator = new QueryParserLocator(new ParserLocationError());
+$buddy = new Buddy($clAddrPort, $parserLocator, new StatementBuilder(), new BuddyRequestError());
 
-$socket = setup();
-
-file_put_contents('/tmp/test.txt', "\n");
 // listening socket in loop
 while (true) {
-	$conn = socket_accept($socket);
-	if ($conn === false) {
-		if (!check_client($pid, $pidPath)) {
-			send_die_msg('Client is dead');
+	if ($socketHandler->hasMsg() === false) {
```

Comment:
Prefer simplicity instead of strict comparision. Can do just `!$socketHandler->hasMsg()`